parties/githubclient.go

```diff
@@ -149,77 +147,69 @@ func (g *GithubOp) RequestTokens(cicHash string) ([]byte, error) {
 	return []byte(jwt.Value), err
 }
 
-func (g *GithubOp) VerifyPKToken(pkt *pktoken.PKToken, cosPk crypto.PublicKey) (map[string]any, error) {
-	if !pkt.OpSigGQ {
-		return nil, fmt.Errorf("non-GQ signatures not supported for github")
+func (g *GithubOp) VerifyPKToken(pkt *pktoken.PKToken, cosPk crypto.PublicKey) error {
+	sigType, ok := pkt.ProviderSignatureType()
+	if !ok {
+		return fmt.Errorf("provider signature type missing")
+	}
+
+	if sigType != pktoken.Gq {
+		return fmt.Errorf("github only support gq signatures")
 	}
 
-	cicphJSON, err := util.Base64DecodeForJWT(pkt.CicPH)
+	cic, err := pkt.GetCicValues()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	expectedAud := string(util.B64SHA3_256(cicphJSON))
+	commitment, err := cic.Commitment()
+	if err != nil {
+		return err
+	}
 
-	idt := pkt.OpJWSCompact()
+	idt, err := pkt.Compact(pkt.Op)
```

Comment:
Should this be the package (pktoken), struct (PKToken) or the instance (pkt)? It feels like the package is best, but I can see the case for the struct. The instance makes me wonder what the particular values in the instance are influencing this output. If this is just a static method why use the instance?

Reply:
This is an object method because it grabs the payload from the object to put into the compact form.

Reply:
It confused me because an instance variable is being supplied to an instance method of the same object. That said, now that I understand what you are doing I withdraw my objection. It is not how I would write it but I see the benefit of flexibility it offers.

Reply:
Resolve after reading